neuraxle/checkpoints.py

```diff
@@ -596,9 +598,302 @@ def __init__(self):
+        )
+
+    def save_one_checkpoint(self, context: ExecutionContext, current_id: str, data_input: Any, expected_output: Any):
+        """
+        Save one expected output data checkpoint with :class:`BaseDataCheckpointer`.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :param current_id: current id
+        :type current_id: str
+        :param data_input: data input
+        :type data_input: Any
+        :param expected_output: expected output
+        :type expected_output: Any
+        :return:
+        """
+        self.data_checkpointer.save_checkpoint(
+            checkpoint_path=self._get_checkpoint_path(context),
+            current_id=current_id,
+            data=expected_output
+        )
+
+    def read_one_checkpoint(self, context: ExecutionContext, current_id: str, data_input: Any, expected_output: Any) -> Tuple:
+        """
+        Read an expected output checkpoint with a :class:`BaseDataCheckpointer`.
+
+        :param context: execution context
+        :type context: ExecutionContext
+        :param current_id: current id
+        :type current_id: str
+        :param data_input: data input
+        :type data_input: Any
+        :param expected_output: expected output
+        :type expected_output: Any
+        :return: current_id, data_input, expected_output
+        :rtype: Tuple[str, Any, Any]
+        """
+        checkpoint = self.data_checkpointer.read_checkpoint(
+            checkpoint_path=self._get_checkpoint_path(context),
+            current_id=current_id
+        )
+
+        return current_id, data_input, checkpoint
+
+class FullCheckpoint(Pipeline):
+    """
+    Full checkpoint pipeline that handles every execution mode.
+
+    #. Saves data checkpoints for data inputs.
+    #. Saves data checkpoints for expected outputs.
+    #. Saves fitted step
+
+    .. seealso::
+        * :class:`Pipeline`
+        * :class:`TransformOnlyWrapper`
+        * :class:`FitOnlyWrapper`
+        * :class:`FitTransformOnlyWrapper`
+        * :class:`StepSavingCheckpointer`
+    """
+    def __init__(self):
+        Pipeline.__init__(self, [
+            TransformOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer())),
+            TransformOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer())),
+
+            FitOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer())),
+            FitOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer())),
+
+            FitTransformOnlyWrapper(DataInputCheckpointerWrapper(PickleDataCheckpointer())),
+            FitTransformOnlyWrapper(ExpectedOutputCheckpointerWrapper(PickleDataCheckpointer())),
+
+            StepSavingCheckpointer(),
+        ])
+
+#
+# TODO: make mini bath sequential pipeline resumable to use this
+# class CheckpointJoiner(MetaStepMixin, Joiner):
+#     def __init__(self, batch_size: int, checkpoint_pipeline):
```

Comment:
Perhaps that this class could be just this: 
```
class DefaultJoiningCheckpointer(Pipeline): 
     def __init__(self): 
        Pipeline.__init__(self, [
            Joiner(),
            SomeDefaultCheckpointClass()
        ])
```

Reply:
that would not work with the existing `MiniBatchSequentialPipeline`... the logic wouldn't detect the Joiner Barrier inside the pipeline. changes would be needed to make your suggestion work, and those changes would be awkward @guillaume-chevalier

Reply:
Okay. Do we really need a `class CheckpointJoiner` as in this TODO then? I'd keep that on the backburner for now, to see later.